hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/operations/GenericOperation.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.ringbuffer.impl.operations;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.ringbuffer.impl.RingbufferContainer;
+
+import java.io.IOException;
+
+import static com.hazelcast.ringbuffer.impl.RingbufferDataSerializerHook.GENERIC_OPERATION;
+
+public class GenericOperation extends AbstractRingBufferOperation {
+
+    public static final byte OPERATION_SIZE = 0;
```

Comment:
Then, I am OK. In fact, I didn't suggest using an enum only for making it visible. That byte variable is used in run() method of the GenericOperation class to determine a specific behaviour using a switch statement. I think we do that a lot in HZ and over-use switch statements for that purpose. We are transferring a value from someplace to another to pick a specific behaviour at some point. I like the good old strategy pattern where we can move the actual behaviour that is going to be executed.  An enum is very useful for such a use-case. We can do it within a method of the enum. For this code, the only changing thing in run() method is the type of the operation that will be called on ringbuffer an it can be easily moved into methods to instances of an enum. Maybe it is a personal thing but I don't like using switch / if-else blocks instead of a strategy object.

Anyway, I just wanted to state my point and I do not want to go beyond the scope of this PR. Kudos for the good work and clean code.